app/src/schemas.ts

```diff
@@ -0,0 +1,8 @@
+import { z } from 'zod'
+
+export const supportedChain = z.enum([
+  'mainnet',
+  'goerli',
+  'sepolia',
+  'optimism',
+])
```

Comment:
As suggested somewhere else, can't you just use `Object.keys(chainConfig)` here instead of redefining the list (which will potentially mismatch the config)?

Reply:
> which will potentially mismatch the config

If this happens ts would throw error as the type doesn't match as config's record key type is this enum value.
But as we don't have typechecking or ci here I'll remove it only consider it in config :D
Would be better for push changes quickly!